rx_signal_processing/dsp.cu

```diff
@@ -57,29 +57,17 @@ namespace {
   }
 
 
-  void frerking_phase_correction(cuComplex *samples, uint32_t num_samps_per_antenna,
-                                  uint32_t num_antennas, double F_s, double F_new, double F_final,
-                                  std::vector<double> freqs)
-  {
-    auto m = F_new / F_final;
-
-    for (uint32_t freq_index=0; freq_index < freqs.size(); freq_index++) {
-      for (int i=0; i<num_antennas; i++) {
-        for (int j=0; j<num_samps_per_antenna; j++){
-          auto phi_k = 2 * M_PI * (F_s/F_new) * fmod((m*j),F_s) * (freqs[freq_index]/F_s);
-          auto phase = std::exp(std::complex<float>(0,1) * std::complex<float>(phi_k,0));
-          cuComplex cu_phase;
-          cu_phase.x = phase.real();
-          cu_phase.y = phase.imag();
-          auto sample_index = freq_index*num_antennas*num_samps_per_antenna + i*num_samps_per_antenna + j;
-          auto corrected_samp = cuCmulf(samples[sample_index],cu_phase);
-          samples[sample_index] = corrected_samp;
-        }
-      }
-    }
-
-  }
-
+  /**
+   * @brief      Drops samples contaminated by edge effects and filter roll off.
+   *
+   * @param      input_samples    The input samples.
```

Comment:
meh. Good that you have a docstring, but 'the input samples' doesn't really expand upon the variable name. What's the type? How are they organized? Number of antennas - main? Int? botH?

Can also add a bit more in the brief that says when it's called, or when it should be called.